sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java

```diff
@@ -0,0 +1,526 @@
+            }
+
+        } else {
+            decryptAndSerializeValue(encryptionSettings, objectNode, propertyValueHolder, propertyName);
+        }
+    }
+
+    public JsonNode decryptAndSerializeValue(EncryptionSettings encryptionSettings, ObjectNode objectNode,
+                                             JsonNode propertyValueHolder, String propertyName) throws MicrosoftDataEncryptionException, IOException {
+        byte[] cipherText;
+        byte[] cipherTextWithTypeMarker;
+
+        cipherTextWithTypeMarker = propertyValueHolder.binaryValue();
+        cipherText = new byte[cipherTextWithTypeMarker.length - 1];
+        System.arraycopy(cipherTextWithTypeMarker, 1, cipherText, 0,
+            cipherTextWithTypeMarker.length - 1);
+        byte[] plainText = encryptionSettings.getAeadAes256CbcHmac256EncryptionAlgorithm().decrypt(cipherText);
+        if (objectNode != null && !objectNode.isNull()) {
+            objectNode.set(propertyName, toJsonNode(plainText,
+                TypeMarker.valueOf(cipherTextWithTypeMarker[0]).get()));
+        }
+        return toJsonNode(plainText,
+            TypeMarker.valueOf(cipherTextWithTypeMarker[0]).get());
+    }
+
+    public static Pair<TypeMarker, byte[]> toByteArray(JsonNode jsonNode) {
+        try {
+            SqlSerializerFactory sqlSerializerFactory = new SqlSerializerFactory();
+            switch (jsonNode.getNodeType()) {
+                case BOOLEAN:
+                    return Pair.of(TypeMarker.BOOLEAN,
+                        sqlSerializerFactory.getDefaultSerializer(Boolean.FALSE).serialize(jsonNode.asBoolean()));
+                case NUMBER:
+                    if (jsonNode.isInt() || jsonNode.isLong()) {
+                        return Pair.of(TypeMarker.LONG,
+                            sqlSerializerFactory.getDefaultSerializer(0l).serialize(jsonNode.asLong()));
+                    } else if (jsonNode.isFloat() || jsonNode.isDouble()) {
+                        return Pair.of(TypeMarker.DOUBLE,
+                            sqlSerializerFactory.getDefaultSerializer(0d).serialize(jsonNode.asDouble()));
+                    }
+                    break;
+                case STRING:
+                    if (jsonNode.asText().length() > STRING_SIZE_ENCRYPTION_LIMIT) {
+                        LOGGER.error("{} length is greater than allowed encryption string length {}",
+                            jsonNode.asText(), STRING_SIZE_ENCRYPTION_LIMIT);
+                    }
+
+                    return Pair.of(TypeMarker.STRING,
+                        SqlSerializerFactory.getOrCreate("varchar", STRING_SIZE_ENCRYPTION_LIMIT, 0, 0).serialize(jsonNode.asText()));
+                case OBJECT:
+                    return Pair.of(TypeMarker.OBJECT,
+                        SqlSerializerFactory.getOrCreate("varchar", STRING_SIZE_ENCRYPTION_LIMIT, 0, 0).serialize(jsonNode.toString()));
+                case ARRAY:
+                    return Pair.of(TypeMarker.ARRAY,
```

Comment:
We must not have these; by the time it comes here, we should only have primitives.

Reply:
Removed it , it was redundant.